include/EASTL/internal/red_black_tree.h

```diff
@@ -184,6 +184,22 @@ namespace eastl
 		rbtree_iterator& operator--();
 		rbtree_iterator  operator--(int);
 
+	protected:
+		template <typename Tp, typename PointerA, typename ReferenceA, typename PointerB, typename ReferenceB>
+		friend bool operator==(const rbtree_iterator<Tp, PointerA, ReferenceA>& a, 
```

Comment:
Also, why do you need to add these comparison operators?  Changing access control doesn't appear to warrant it.